### DISCUSSION TITLE
Setup with M1

### DISCUSSION THREAD

**Discussion by FunnyGhost** [eieioxyz/dotfiles_macos#23] [Q&A]
Hello,

I recently purchased an Apple MacBook Pro M1 and had a chance to put my dotfiles setup to the test. Everything went smooth! However, it installs the Intel version of apps. For example **vscode**, **webstorm**, these I had to uninstall and manually install the M1 version of them(download from their website). Any idea on how to automate this part?

Thanks a lot!

**Comment by eieioxyz** [comment]
Congrats on the M1! I don't have one to test on, but it looks like you just need to play the waiting game before those will be part of your automated process.

Here's an [issue on VS Code's repo](https://github.com/Homebrew/homebrew-cask/issues/100705) and [here's a PR that should resolve it](https://github.com/Homebrew/homebrew-cask/pull/98725). I don't know their schedule for getting that code into production.

I didn't investigate Webstorm, but I assume it's a similar issue. [This explains](https://www.ithinkdiff.com/homebrew-apple-silicon-m1-macs/) that all the "bottles" are in the process of being converted to work with M1.

**Reply by FunnyGhost** [reply]
Thanks for the reply! So, if I use brew in a **rosetta terminal**, it installs the intel version of apps. However, if I use brew in the **normal terminal**, it installs the M1 version of everything it finds. Pretty interesting 🤔. (or maybe it's just a coincidence)

Long story short: everything works nice and smooth now. Thanks!

P.S.: Can't wait for the "node setup" part of the course 😁
